### PR TITLE
Update plugin

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
* detach ancient Sonatype parent POM (lift whatever needed into project POM)
* update plugins to latest
* prerequisite (to run the plugin): 3.2.5
* prerequisite (to build the plugin): 3.2.5 (but I'd move this to 3.8.6 or so, but minimum is 3.2.5 as Maven lifted "minimum" Maven version to 3,2,5 on latest plugins)
* update minor of some deps

After this change release/deploy should works as before (did not test naturally).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -76,9 +70,41 @@
...
+        <enabled>true</enabled>
+      </snapshots>
+    </repository>
+  </repositories>
+
+  <distributionManagement>
+    <snapshotRepository>
+      <id>sonatype-nexus-snapshots</id>
+      <name>Sonatype Nexus Snapshots</name>
+      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
```
Comment (kwin) [inline pom.xml:91]: Shouldn't we rather use Eclipse's repository (https://github.com/orgs/eclipse/teams/technology-sisu-committers/discussions/1/comments/3, https://wiki.eclipse.org/Services/Nexus) as there are secrets on Eclipse Jenkins and optionally in GitHub secrets to push it there? I would like to deploy SNAPSHOTs with every `master` CI build. Using OSSRH would require to use a personal access token in GitHub secrets (bound to a specific user)

Comment (cstamas) [reply pom.xml:91]: I agree, but this PR does not change anything in this respect, it merely removes ancient sonatype parent POM (and moves relevant bits to POM). Hence, I agree with your proposal, but IMO that is for another PR.

Comment (kwin) [reply pom.xml:91]: Still not sure if we shouldn't rather introduce a new parent in `sisu.inject` instead which is leveraged by `sisu.mojos` as some parts are definitely shared among those.

Comment (cstamas) [reply pom.xml:91]: Yes, IMO we should make one of the POMs as parent for others as well, but again, baby steps...

**Review by kwin — APPROVED** [review]

**Pull request closed without merging**